internal/p2p/pex/reactor.go

```diff
@@ -291,6 +302,7 @@ func (r *Reactor) processPeerUpdate(peerUpdate p2p.PeerUpdate) {
 	switch peerUpdate.Status {
 	case p2p.PeerStatusUp:
 		r.availablePeers[peerUpdate.NodeID] = struct{}{}
+		r.recentlyAvailablePeers.Add(peerUpdate.NodeID, struct{}{})
 	case p2p.PeerStatusDown:
```

Comment:
Shall we remove them from the cache once the peer is down?

Reply:
the idea is to keep it so we can retry once we've exhausted all peers. So let's say we have the following peers:
A, B, C
after a while, due to p2p / intermittnet network issues, A, B, C slowly get marked as unhealty
We're now in a situation where we have no available peers. Current logic exits p2p https://github.com/sei-protocol/sei-tendermint/blob/main/internal/p2p/pex/reactor.go#L177 and the node ends up hanging (which is what we saw on snapshot node)
This logic sets available peers back to ABC and retries them.
The LRU nature of the cache will slowly evict older ones